src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
+      ;; because :JRubyMetricsService isn't in bootstrap.cfg.
+      (assoc context :request-handler (if (> max-active-requests 0)
```

Comment:
Why would we want to tie this to the `max-active-requests` specifically?  Where `max-active-requests` indicates the number of requests which can be made to a JRuby instance before it is recycled, it seems like the values for it could range wildly.  Ideally, users would be able to set this to 0 (or to a very high number) for best performance - i.e., for the case where there are no memory leaks.  It seems like in either of these cases that there wouldn't be much of a benefit to the extra code added here.